benchmark/benchmark.py

```diff
@@ -0,0 +1,150 @@
+import argparse
+import json
+from itertools import chain
+from random import sample, shuffle
+from time import perf_counter
+from typing import List, Optional, Tuple
+
+import pandas as pd
+import seaborn as sns
+from openvino import CompiledModel, compile_model
+from openvino_tokenizers import convert_tokenizer
+from tqdm.auto import tqdm
+from transformers import AutoTokenizer, PreTrainedTokenizerBase
+
+
+def sample_texts(
+    dataset_path: str,
+    num_texts: int = 1000,
+) -> List[Tuple[str, str]]:
+    with open(dataset_path) as f:
+        dataset = json.load(f)
+    # Filter out the conversations with less than 2 turns.
+    dataset = [data for data in dataset if len(data["conversations"]) >= 2]
+    # Only keep the first two turns of each conversation.
+    dataset = [
+        (data["conversations"][0]["value"], data["conversations"][1]["value"]) for data in sample(dataset, k=num_texts)
+    ]
+    shuffle(dataset)
+    return dataset
+
+
+def benchmark_tokenizers(
+    ov_tokenizer: CompiledModel,
+    hf_tokenizer: PreTrainedTokenizerBase,
+    dataset: List[Tuple[str, str]],
+):
+    columns = ["prompt", "OV", "HF"]
+    results = []
+
+    # warmup
+    for repeat in range(1, 11):
+        ov_tokenizer(["test " * repeat])
+        hf_tokenizer(["test " * repeat])
+
+    for prompt in tqdm(chain.from_iterable(dataset), total=len(dataset) * 2):
+        res = [prompt]
+
+        ov_start = perf_counter()
+        ov_tokenizer([prompt])
+        res.append(perf_counter() - ov_start)
+
+        hf_start = perf_counter()
+        hf_tokenizer(prompt)
+        res.append(perf_counter() - hf_start)
+
+        results.append(res)
+
+    return pd.DataFrame(results, columns=columns)
+
```

Comment:
can we have breakdown for OpenVINO operations?

Reply:
currently, it takes stat only from last inference 
let's have cumulative statistic

Reply:
```shell
                     count        mean         std      min        50%          max
Prompt Length         2000 1060.864000 1413.245877 1.000000 545.000000 15463.000000
# Tokens              2000  234.236500  322.702295 1.000000 117.000000  3149.000000
BPETokenizer_87       2000    0.002023    0.001810 0.001417   0.001548     0.027062
RegexSplit_79         2000    0.000214    0.000071 0.000184   0.000194     0.001148
BytesToChars_80       2000    0.000050    0.000001 0.000049   0.000050     0.000057
StringTensorUnpack_2  2000    0.000019    0.000001 0.000019   0.000019     0.000027
RaggedToDense_96      2000    0.000019    0.000000 0.000019   0.000019     0.000023
Convert_136           2000    0.000004    0.000000 0.000004   0.000004     0.000005
Range_15              2000    0.000002    0.000000 0.000002   0.000002     0.000004
Convert_97_1          2000    0.000002    0.000000 0.000002   0.000002     0.000002
RaggedToDense_96.0    2000    0.000001    0.000000 0.000001   0.000001     0.000002
Range_9               2000    0.000001    0.000000 0.000001   0.000001     0.000005
ReduceMax_94          2000    0.000001    0.000000 0.000001   0.000001     0.000001
Add_13                  17    0.000001    0.000000 0.000001   0.000001     0.000001
```